include/dali/core/mm/pool_resource.h

```diff
@@ -59,11 +86,159 @@ constexpr pool_options default_device_pool_opts() noexcept {
   return { (static_cast<size_t>(1) << 32), (1 << 20), 2.0f, true, true };
 }
 
+template <memory_kind kind>
+constexpr sync_scope default_sync_scope() {
+  return kind == memory_kind::device ? sync_scope::device
+                                     : kind == memory_kind::host ? sync_scope::none
+                                                                 : sync_scope::system;
+}
+
+template <memory_kind kind>
+constexpr pool_options default_pool_opts() noexcept {
+  if (kind == memory_kind::host) {
+    return default_host_pool_opts();
```

Comment:
Why `opt.sync` is not set for symmetry?

Reply:
Because there's only one `host` kind and the default is already set by default_**host**_pool_opts - for device, it depends on the specific kind, so we have to change it.